idunn/datasources/pages_jaunes.py

```diff
@@ -58,21 +77,78 @@ def get_places_bbox(self, raw_categories, bbox, size=10, query=""):
         raw_places = result.get("hits", {}).get("hits", [])
         return [PjPOI(p["_source"]) for p in raw_places]
 
-    def get_place(self, id):
-        internal_id = id.replace(f"{self.PLACE_ID_NAMESPACE}:", "", 1)
-
+    def get_place(self, poi_id):
         es_places = self.es.search(
             index=self.es_index,
-            body={"query": {"bool": {"filter": {"term": {"_id": internal_id}}}}},
+            body={"query": {"bool": {"filter": {"term": {"_id": self.internal_id(poi_id)}}}}},
             ignore_unavailable=True,
         )
 
         es_place = es_places.get("hits", {}).get("hits", [])
         if len(es_place) == 0:
-            raise HTTPException(status_code=404, detail=f"place {id} not found")
+            raise HTTPException(status_code=404, detail=f"place {poi_id} not found")
         if len(es_place) > 1:
-            logger.warning("Got multiple places with id %s", id)
+            logger.warning("Got multiple places with id %s", poi_id)
         return PjPOI(es_place[0]["_source"])
 
 
-pj_source = PjSource()
+class ApiPjSource(PjSource):
+    PJ_INFO_API_URL = "https://api.pagesjaunes.fr/v1/pros"
+    PJ_FIND_API_URL = "https://api.pagesjaunes.fr/v1/pros/search"
+    PJ_AUTHORIZATION = "https://api.pagesjaunes.fr/oauth/client_credential/accesstoken"
+
+    def __init__(self):
+        self.enabled = True
+        self.client_id = settings.get("PJ_API_ID")
+        self.client_secret = settings.get("PJ_API_SECRET")
+
+        # TODO: refresh token if available?
+        #       https://requests-oauthlib.readthedocs.io/en/latest/oauth2_workflow.html#refreshing-tokens
+        self.client = BackendApplicationClient(client_id=self.client_id)
+        self.oauth = OAuth2Session(client=self.client)
+        self.token = self.oauth.fetch_token(
+            token_url=self.PJ_AUTHORIZATION,
+            client_id=self.client_id,
+            client_secret=self.client_secret,
+        )
```

Comment:
I'll have a look into it.

I am not sure PJ doesn't intend to provide a refresh token in the future as several fields related to it are provided (`refresh_token_expires_in` and `refresh_count`) and mostly because the [quickstart](https://developer.pagesjaunes.fr/quickstart) page of the documentation has a section "Refresh access token" in the table of contents which doesn't seem to be filled yet. I guess requesting a new token by hand is very simple and clean enough even if an alternative mechanism would be provided?